pkg/network/tracer/connection/tcp_close_consumer.go

```diff
@@ -75,101 +72,47 @@ func (c *tcpCloseConsumer) Stop() {
 	if c == nil {
 		return
 	}
-	c.eventHandler.Stop()
 	c.once.Do(func() {
 		close(c.closed)
 	})
 }
 
-func (c *tcpCloseConsumer) extractConn(data []byte) {
-	ct := (*netebpf.Conn)(unsafe.Pointer(&data[0]))
-	conn := c.buffer.Next()
-	populateConnStats(conn, &ct.Tup, &ct.Conn_stats, c.ch)
-	updateTCPStats(conn, &ct.Tcp_stats)
+func (c *tcpCloseConsumer) Callback(conn *network.ConnectionStats) {
+	// sentinel record post-flush
+	if conn == nil {
+		request := <-c.flushChannel
```

Comment:
Any chance this would block? Or will we will only get a nil connection here if Flush() is called?

Reply:
If there is nothing to flush do we still get a sentinel value?